### ISSUE TITLE
Sidekick doesn't work after some time

### ISSUE THREAD

**Issue by jezznar** [domialex/Sidekick#276]
After using sidekick then playing for quite some time(around > 1hr), when I used sidekick again it doesn't work. I have to restart it to make it work again.

**Comment by domialex** [comment]
That's going to be hard to reproduce, I've had mine running since yesterday.
When you say that it doesn't work, you mean just the hotkeys? When you press a hotkey can you see it the logs capture it?

**Comment by ba-chan** [comment]
It can be a lot less than an hour, I've had it fail after starting up and running a single map.

I've just had it happen again, exited a map and found CTRL+D (I remapped Check Prices) unresponsive.  I looked at the log and it had registered the key presses for the action, but it displayed nothing, and would not log any more key presses until restarted.

I can't remember it happening with v0.5.0 but that was during league start I wasn't trading too much, but I certainly noticed it soon after installing v0.5.1, and it's still dong it with v0.5.2.  

I'm running Win10 pro, with the game in full screen windowed mode on a multiple monitor setup, so I tend to be switching between poe and browsers, libreoffice, etc, quite a lot, so might be related to that.  If I spot a pattern I'll post it here.

**Comment by jezznar** [comment]
do we have logs that we can check or any crash dump anywhere? I don't see any logs in the sidekick folder

**Comment by leMicin** [comment]
We currently do not log to any files. This is a feature we should definetly consider.

**Comment by ba-chan** [comment]
I had a look at the log after it happened again, confirming that the initial keypress is logged, eg "16/03/2020 08:16:13 +00:00 - Keybind Triggered - Check Prices", but nothing else is logged after that, and further keypresses for any function, such as return to hideout too, fail to do anything.

However, I see the comment above about stuttering.  I'm not seeing this, except for just the once a day or so ago, when I did a price check and my computer stuttered like crazy for several seconds.

**Comment by jezznar** [comment]
> However, I see the comment above about stuttering. I'm not seeing this, except for just the once a day or so ago, when I did a price check and my computer stuttered like crazy for several seconds.

Yes this is the symptom as mine

**Comment by streetmapp** [comment]
I haven't had this work since v0.5.1 came out. v0.5.0 was working fine. Then with these updates haven't had it work. I check the logs and it registers my key press, but nothing happens.

```
3/16/2020 9:24:40 AM -04:00 - Fetching PoeNinja cache.
3/16/2020 9:24:40 AM -04:00 - PoeNinja cache fetched.
3/16/2020 9:24:40 AM -04:00 - 70% - Initializer - End After Init (PoeNinjaCache)
3/16/2020 9:24:40 AM -04:00 - 70% - Initializer - Start After Init (KeybindEvents)
3/16/2020 9:24:40 AM -04:00 - 80% - Initializer - End After Init (KeybindEvents)
3/16/2020 9:24:40 AM -04:00 - 80% - Initializer - Start After Init (NativeKeyboard)
3/16/2020 9:24:40 AM -04:00 - 90% - Initializer - End After Init (NativeKeyboard)
3/16/2020 9:24:40 AM -04:00 - 90% - Initializer - Start After Init (NativeProcess)
3/16/2020 9:24:40 AM -04:00 - 100% - Initializer - End After Init (NativeProcess)
3/16/2020 9:25:18 AM -04:00 - Permission Sufficient.
3/16/2020 9:25:48 AM -04:00 - Keybind Triggered - Check Prices
```

**Comment by ba-chan** [comment]
Regarding the stuttering, I can reproduce this 100% by price checking an item, hitting space to dismiss the panel, and immediately price checking the same item again.  Another panel will pop up telling me "Unable to find prices for this item" for a few seconds, then it vanishes and my pc stutters for several seconds until Sidekick terminates.

Looks like we're discussing two separate bugs here, as the stutter is followed by app termination (although the icon remains in system tray utill hovered over) rather than just failing to process input.

**Comment by jezznar** [comment]
Actually, just now I discovered that price checking without hovering to an item causes "Unable to find prices for this item" then stuttering starts.

**Comment by ba-chan** [comment]
Oh, this is interesting.  If I do a price check on an empty space I don't get the stutter, but it does result in the problem of input locking up.

**Comment by leMicin** [comment]
Thanks for the report guys, we are working on a completely new parser to support advanced search and item filtering. Later this week we are going to release 0.6.0. We will make sure this problem does not happen again.

**Comment by domialex** [comment]
Just to add a reference, I can reproduce the same behaviour by price checking two items without closing the first overlay.

**Comment by Blinke** [comment]
I figured out why this is happening, at least when price checking anything that is not an item (empty space, etc).

In BaseWindow we're manually calling `Close()` on the in `KeybindEvents_OnCloseWindow`. At this point the "error" window is already in the process of closing, which throws an exception and crashes the program.

**Comment by leMicin** [comment]
I can't seem to be able to reproduce this issue. This is the last issue remaining for a 0.6.0 release.

**Comment by leMicin** [comment]
Never mind, I got it. It should be fixed for 0.6.0. Thanks for the help everyone!
